File: Backend/Pricing.Application/Domain/Entities/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Baldan.Pricing.Application.Domain.Entities
{
    public class Product : EntityBase
    {
        public string Name { get; set; }
        public string? Description { get; set; }
        public double Price_Unit { get; set; }
        public double Origin_Price { get; set; }
        public int Id_Category { get; set; }
        public Product_Category Product_Category { get; set; }
        public double? Rating { get; set; }
        public int? ReviewCount { get; set; }
        public int CountSold { get; set; }
        public int Total_Stock { get; set; }
        public string? Badge { get; set; }
        public bool FreeShipping { get; set; }
        public int? installments { get; set; }
        public string Tags { get; set; }
        public bool? Featured { get; set; }
        public  string? Cod_Cupom { get; set; }
        public ICollection<Imagens_Products> Imagens_Products { get; set; } = [];
        public ICollection<Variante_Products> Variante_Products { get; set; } = [];
        public ICollection<Cart_Item> Cart_Items { get; set; } = [];
        public ICollection<OrderLineItens> OrderLineItens { get; set; } = [];

        public ICollection<Rating> Ratings { get; set; } = [];

    }
}

File: src/models/Product.ts
import { UserRanting } from "./User";

export interface Product {
    id?: number;
    name: string;
    description: string;
    price_Unic: number;
    origin_Price?: number;
    imagens: Imagens_Products[];
    id_category: number;
    review_Count: number;
    count_Sold: number;
    variations: ProductVariation[];
    total_Stock: number;
    badge?: string;
    freeShipping: boolean;
    installments?: number;
    tags: string;
    featured: boolean;
    insertDate: Date;
    ativo?: boolean;
    showBanner?: boolean;
    valorDicont?: number;
    rating: number;

}

export interface ProductAdmin {
    id: number;
    name: string;
    description: string;
    price_Unic: number;
    origin_Price?: number;
    imagens: Imagens_Products[];
    id_category: number;
    rating: number;
    review_Count: number;
    count_Sold: number;
    variations: ProductVariation[];
    total_Stock: number;
    badge?: string;
    freeShipping: boolean;
    installments?: number;
    tags: string;
    featured: boolean;
    insertDate: Date;
    removedImages: number[];
    removedVariants: number[];
    ativo?: boolean;
    showBanner?: boolean;
}

export interface ProductSaveOrder {
    id: number;
    name: string;
    price_Unic: number;
    quantity: number;
    imagens?: Imagens_Products[];
    freeShipping?: boolean;
    category?: string;
    origin_Price?: number;
    badge?: string;
    Count_Rating?: number;
    variations: ProductVariation | null;
    id_category?: number;
    valorDicont?: number;
    evaluated?: boolean

}
export interface ProductVariation {
    id: number;
    id_Product?: number;
    name: string;
    value: string;
    type: string;
    stoke: number;
    price_Modifier?: number;
    new?: boolean;
}
export type ProductCategory = 'eletronicos' | 'garrafas' | 'acessorios' | 'virais';

export interface Imagens_Products {
    id: number;
    id_Product: number;
    url_Imagem: string;
    file?: File;
}
export interface Category {
    id: number;
    category: string;
    description: string;
    banners?: string | string[];
    imagem: string;
    color: string;
    meta_Title: string;
    meta_Description: string;
    ativo: boolean;
    insertDate?: Date;

}
export interface RantingAllProduct {
    id: number;
    id_Product: number;
    id_User_Customer: number;
    ranting: number;
    comment: string;
    media: string;
    recomend: Boolean;
    insertDate: Date;
    updateDate: Date;
    isDelete: boolean;
    user: UserRanting;
}